compiler/rustc_parse/src/parser/mod.rs

```diff
@@ -1514,6 +1516,75 @@ impl<'a> Parser<'a> {
         Ok(())
     }
 
+    /// Parses `kw`, `kw(in path)` plus shortcuts `kw(crate)` for `kw(in crate)`, `kw(self)` for
+    /// `kw(in self)` and `kw(super)` for `kw(in super)`.
+    fn parse_restriction(
+        &mut self,
+        kw: ExpKeywordPair,
+        feature_gate: Option<Symbol>,
+        fbt: FollowedByType,
+    ) -> PResult<'a, Restriction> {
+        if !self.eat_keyword(kw) {
+            // We need a span, but there's inherently no keyword to grab a span from for an implied
+            // restriction. An empty span at the beginning of the current token is a reasonable
+            // fallback.
+            return Ok(Restriction::implied().with_span(self.token.span.shrink_to_lo()));
```

Comment:
Can we just make `Restriction::implied` take the span directly?
```suggestion
            return Ok(Restriction::implied(self.token.span.shrink_to_lo()));
```